### PR TITLE
User table

### PR DESCRIPTION
**Pull request by DeepikaGhodki**
### **Changes**

1. Endpoints to list all users and update them (only admin status update supported as now)
2. A new route `/users` that lists all users present in the system
3. This page is only accessible to an admin user
4. Only admin status of a given user can be changed using the toggle switch (in admin column) 

**To-do**

- [x] Add pagination to list all users
- [x] Client implementation of update admin status

**User Page**

Screenshot from 2022-06-29 14-18-37

### CODE REVIEW

File: nmdc_server/api.py [hunk 1]
```diff
@@ -552,3 +552,28 @@ async def submit_metadata(
...
+
+
+@router.get(
+    "/users", responses=login_required_responses, response_model=List[schemas.User], tags=["user"]
+)
+async def get_users(
+    db: Session = Depends(get_db),
+    user: models.User = Depends(admin_required),
+):
+    users = db.query(User).all()
```
Comment (DeepikaGhodki) [inline nmdc_server/api.py:564]: @subdavis FYI - given the volume of users in not high, we're fetching all rows at once as of now. Would you recommend introducing pagination parameters here?

Comment (subdavis) [reply nmdc_server/api.py:564]: Yeah, I think server-side pagination is a good idea here.   I need to add it to the submission list as well.

File: web/src/data/api.ts [hunk 2]
```diff
@@ -309,6 +309,13 @@ export interface BinResponse<T = string | number> {
   facets: number[];
 }
 
+export interface User{
+    id: string,
+    orcid: string;
+    name: string;
+    isAdmin: boolean;
```
Comment (subdavis) [inline web/src/data/api.ts:316]:
```suggestion
    is_admin: boolean;
```

**Review by subdavis — CHANGES_REQUESTED** [review]
This looks really good!  Aside from a couple comments, please see my pagination adaptation in #734 and do something similar here.

💯

File: web/src/views/User/UserPage.vue [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    <v-container>
+      <v-card flat>
+        <v-card-title class="text-h4">
+          Users
+        </v-card-title>
+        <v-card outlined>
+          <v-data-table
+            dense
+            :headers="headers"
+            :items="users.data.results"
```
Comment (subdavis) [inline web/src/views/User/UserPage.vue:57]:
```suggestion
            :items="users.data.results.results"
```

**Review by subdavis — CHANGES_REQUESTED** [review]
Small issue

File: web/src/views/User/UserPage.vue (hunk 3 above)
Comment (DeepikaGhodki) [reply web/src/views/User/UserPage.vue]: Oh thanks!

**Review by subdavis — APPROVED** [review]
:100: Awesome

**Comment by ssarrafan** [comment]
completed in July

**Pull request merged**